lib/gui/lvgl/lvgl.c

```diff
@@ -146,6 +149,83 @@ static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 }
 #endif /* CONFIG_LVGL_BUFFER_ALLOC_STATIC */
 
+#ifdef CONFIG_LVGL_KSCAN
+struct kscan_fifo_data {
+	void *fifo_reserved;
+	lv_indev_data_t data;
+};
+
+static K_MEM_SLAB_DEFINE(kscan_slab, sizeof(struct kscan_fifo_data), 10, 4);
+static K_FIFO_DEFINE(kscan_fifo);
```

Comment:
Why not using a k_msgq here?

Reply:
no reason, thanks for the suggestion. i've updated the pr to use a msgq.